src/python/pants/auth/cookies.py

```diff
@@ -0,0 +1,52 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import os
+
+from six.moves.http_cookiejar import LWPCookieJar
+
+from pants.subsystem.subsystem import Subsystem
+from pants.util.dirutil import safe_mkdir_for
+
+
+class BasicAuthException(Exception):
+  pass
+
+
+class Cookies(Subsystem):
+  options_scope = 'cookies'
+
+  @classmethod
+  def register_options(cls, register):
+    super(Cookies, cls).register_options(register)
+    register('--path', advanced=True, default='~/.pants.cookies',
```

Comment:
A change in this option should invalidate the `Login` task.

Reply:
I guess you're right conceptually, although that task doesn't currently consult the invalidation cache at all. It could if we put the cookies under .pants.d, as discussed earlier. I'm now leaning more towards that, I guess. Will think on it for a few minutes.